modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/ExpressionVisitorImpl.java

```diff
@@ -292,61 +302,64 @@ private Filter _getANDFilter(Filter leftFilter, Filter rightFilter) {
 		return booleanFilter;
 	}
 
-	private Filter _getEQFilter(
-		EntityField entityField, Object fieldValue, Locale locale) {
-
-		if (fieldValue == null) {
-			BooleanFilter booleanFilter = new BooleanFilter();
-
-			booleanFilter.add(
-				_getNullValueFilter(entityField, locale),
-				BooleanClauseOccur.MUST_NOT);
-
-			return booleanFilter;
-		}
-
-		return new QueryFilter(
-			_nestedFieldQueryHelper.getQuery(
-				entityField.getFilterableName(locale),
-				fieldName -> new TermQueryImpl(
-					fieldName, entityField.getFilterableValue(fieldValue))));
-	}
-
-	private Optional<Filter> _getFilterOptional(
+	private Optional<Object> _getBinaryExpressionOptional(
 		BinaryExpression.Operation operation, Object left, Object right,
 		Locale locale) {
 
-		Filter filter = null;
+		Object binaryExpression = null;
```

Comment:
This doesn't return a binaryExpression.

It returns a filter or a String.

Could you think in a better name?

Reply:
Done!